### PR TITLE
fix: build timeout validation

### PR DESCRIPTION
**Pull request by AbhishekA1509**
# Description
Adding a validator function to validate required positive integer and move validation texts to a constant.

### CODE REVIEW

File: src/Pages/GlobalConfigurations/BuildInfra/constants.tsx [hunk 1]
```diff
@@ -14,7 +14,7 @@ import {
 } from './types'
 
 export const BUILD_INFRA_INPUT_CONSTRAINTS = {
-    STEP: 0.01,
+    STEP: 'any',
```
Comment (eshankvaish) [inline src/Pages/GlobalConfigurations/BuildInfra/constants.tsx:17]: add a comment describing it

File: src/Shared/validations.tsx [hunk 2]
```diff
@@ -47,22 +60,67 @@ export const validateRequiredPositiveNumber = (value: string | number): Validati
...
+    const numericValue = Number(value)
+
+    if (numericValue > Number.MAX_SAFE_INTEGER) {
+        return {
+            isValid: false,
+            message: MESSAGES.MAX_SAFE_INTEGER,
+        }
+    }
+
+    if (Number(value) <= 0) {
```
Comment (eshankvaish) [inline src/Shared/validations.tsx:84]:
```suggestion
    if (numericValue <= 0) {
```

File: src/Shared/validations.tsx [hunk 3]
```diff
@@ -47,22 +60,67 @@ export const validateRequiredPositiveNumber = (value: string | number): Validati
...
+            message: MESSAGES.VALID_POSITIVE_NUMBER,
+        }
+    }
+
+    return {
+        isValid: true,
+    }
+}
+
+export const validatePositiveInteger = (value: string | number): ValidationResponseType => {
```
Comment (eshankvaish) [inline src/Shared/validations.tsx:96]: this is duplicated in the above function as well. Can we reuse this function in the above one?

File: src/Shared/validations.tsx [hunk 4]
```diff
@@ -47,22 +60,67 @@ export const validateRequiredPositiveNumber = (value: string | number): Validati
...
+    const numericValue = Number(value)
+
+    if (numericValue > Number.MAX_SAFE_INTEGER) {
+        return {
+            isValid: false,
+            message: MESSAGES.MAX_SAFE_INTEGER,
         }
     }
 
     if (Number(value) <= 0) {
```
Comment (eshankvaish) [inline src/Shared/validations.tsx:120]:
```suggestion
    if (numericValue <= 0) {
```

**Review by eshankvaish — CHANGES_REQUESTED** [review]

File: src/Shared/validations.tsx (hunk 4 above)
Comment (AbhishekA1509) [reply src/Shared/validations.tsx]: Done

File: src/Shared/validations.tsx (hunk 2 above)
Comment (AbhishekA1509) [reply src/Shared/validations.tsx]: Done

File: src/Shared/validations.tsx (hunk 3 above)
Comment (AbhishekA1509) [reply src/Shared/validations.tsx]: Messaging and requirements are different

File: src/Pages/GlobalConfigurations/BuildInfra/constants.tsx (hunk 1 above)
Comment (AbhishekA1509) [reply src/Pages/GlobalConfigurations/BuildInfra/constants.tsx:18]: done

**Review by eshankvaish — APPROVED** [review]

**Pull request merged**